### PR TITLE
Performance optimizations for coordinator

### PR DESCRIPTION
**Pull request by cberner**

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/metadata/HandleResolver.java [hunk 1]
```diff
@@ -169,4 +172,112 @@ public ConnectorHandleResolver resolverFor(String id)
...
+            return partitioningHandle;
+        }
+
+        public Optional<Class<? extends ConnectorTransactionHandle>> getTransactionHandleClass()
+        {
+            return transactionHandle;
+        }
+
+        @Override
+        public boolean equals(Object o)
```
Comment (dain) [inline presto-main/src/main/java/com/facebook/presto/metadata/HandleResolver.java]: I don't think equals and hash code are ever used.

Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/metadata/HandleResolver.java]: Yes, definitely remove these

File: presto-main/src/main/java/com/facebook/presto/metadata/HandleResolver.java [hunk 2]
```diff
@@ -40,127 +43,127 @@
...
     public String getId(ConnectorTransactionHandle transactionHandle)
     {
-        return getId(transactionHandle, ConnectorHandleResolver::getTransactionHandleClass);
+        return getId(transactionHandle, MaterializedHandleResolver::getTransactionHandleClass);
     }
 
     public Class<? extends ConnectorTableHandle> getTableHandleClass(String id)
     {
-        return resolverFor(id).getTableHandleClass();
+        return resolverFor(id).getTableHandleClass().get();
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/metadata/HandleResolver.java]: Make all these `orElseThrow` that have a good exception message

File: presto-main/src/main/java/com/facebook/presto/metadata/HandleResolver.java (hunk 1 above)
Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/metadata/HandleResolver.java]: `equals` is used on line 61, and I figured I should implement `hashCode` as well

**Pull request merged**